### PR TITLE
EVG-12471 allow specifying headers for webhook subscriptions

### PR DESCRIPTION
**Pull request by john-m-liu**
Alignment is a bit off but I decided to not mess with css since this is in the new UI anyway
Screen Shot 2020-07-16 at 4 18 35 PM

### CODE REVIEW

**Comment by jimoleary** [comment]
Would it be possible to have a private checkbox for security related token? Something similar to project configuration.

**Comment by john-m-liu** [comment]
What would you want the checkbox to do?

**Comment by jimoleary** [comment]
> What would you want the checkbox to do?

redact the value like on the [evergreen project pages](https://evergreen.mongodb.com/projects##mongodb-mongo-master):

image

**Comment by john-m-liu** [comment]
If you're asking for a way to not display it, that's not really more secure except in the case of someone looking over your shoulder. Private project vars are not sent to the client at all, which means that you can't edit them after setting them. Is that the behavior you want?

**Comment by jimoleary** [comment]
> If you're asking for a way to not display it, that's not really more secure except in the case of someone looking over your shoulder. Private project vars are not sent to the client at all, which means that you can't edit them after setting them. Is that the behavior you want?

Yes, that would be best IMHO.

**Comment by bsamek** [comment]
@jimoleary What would be the use case for that? Are there secrets that only some members of your team should have access to? I think that should be a separate ticket, with an explanation of why that would be useful.

**Comment by jimoleary** [comment]
> @jimoleary What would be the use case for that? Are there secrets that only some members of your team should have access to? I think that should be a separate ticket, with an explanation of why that would be useful.

I suppose if visibilty is limited to a single team then it's not an requirement.

**Comment by john-m-liu** [comment]
@jimoleary what sort of secret are you putting here? I've tried 2 different approaches to this, but both add complexity to the code in ways that I'd like to avoid

**Comment by jimoleary** [comment]
> @jimoleary what sort of secret are you putting here? I've tried 2 different approaches to this, but both add complexity to the code in ways that I'd like to avoid

@john-m-liu It will be along the lines of:

Key: "Authorization" 
Value: "Bearer ${JWT}"

Where JWT is a string of alphanumeric characters (length wise the example that I have is 616 characters long).

See [this comment](https://jira.mongodb.org/browse/DAG-274?focusedCommentId=3276831&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3276831)

**Comment by john-m-liu** [comment]
I messaged Jim and because this data is only accessible by those with project view security, the existing PR is sufficiently secure

File: rest/model/subscriber_test.go [hunk 1]
```diff
@@ -106,8 +106,9 @@ func TestSubscriberModelsWebhook(t *testing.T) {
 	assert := assert.New(t)
 
 	target := event.WebhookSubscriber{
-		URL:    "foo",
-		Secret: []byte("bar"),
+		URL:     "foo",
+		Secret:  []byte("bar"),
+		Headers: []event.WebhookHeader{},
```
Comment (ybrill) [inline rest/model/subscriber_test.go:111]: is this necessary?

Comment (john-m-liu) [reply rest/model/subscriber_test.go:111]: I believe the test hit a nil pointer when this wasn't there

**Pull request merged**